### PR TITLE
Add a "coveralls" target for generating and uploading test coverage.

### PR DESCRIPTION
**Pull request by G-Harmon**
This target installs goveralls (via "go get") and then runs it to generate and upload test coverage data.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/k8s-multicluster-ingress/63)

### CODE REVIEW

**Comment by G-Harmon** [comment]
cc @nikhiljindal @madhusudancs @BenTheElder

File: Makefile [hunk 1]
```diff
@@ -37,5 +37,11 @@ cover:
 	@echo "+ $@"
 	@go list -f '{{if len .TestGoFiles}}"go test -coverprofile={{.Dir}}/.coverprofile {{.ImportPath}}"{{end}}' $(shell go list ${PKG}/... | grep -v vendor) | xargs -L 1 sh -c
 
+coveralls:
+	@echo "+ $@"
+	@go get github.com/mattn/goveralls
```
Comment (nikhiljindal) [inline Makefile]: Add a comment why is this required?

File: Makefile [hunk 2]
```diff
@@ -37,5 +37,11 @@ cover:
...
+# Depends on $COVERALLS_TOKEN (in config.yaml for automated runs)
+	@goveralls
```
Comment (nikhiljindal) [inline Makefile]: FYI for comparison, this is what we use at other places: https://github.com/kubernetes/ingress-nginx/blob/b68feb9c590e2c11bdb098549f3208f80029daf7/Makefile#L151.
Dont we need to pass the repotoken?

Comment (BenTheElder) [reply Makefile]: https://github.com/mattn/goveralls#usage

> You can set the environment variable $COVERALLS_TOKEN to your token so you do not have to specify it at each invocation.

Comment (BenTheElder) [reply Makefile]: `COVERALLS_TOKEN` is being set on the prow-job-container's env in the test-infra PR

File: Makefile (hunk 1 above)
Comment (G-Harmon) [reply Makefile]: done.

File: Makefile (hunk 2 above)
Comment (G-Harmon) [reply Makefile]: It looks like the difference is that ingress-nginx runs "go test" separately, and in doing so, also filters out "test/e2e". Should we do that?

Comment (nikhiljindal) [reply Makefile]: Yes we can filter it out, but not really a blocker for this PR. We can also filter it in the PR that adds that directory.

File: Makefile [hunk 3]
```diff
@@ -37,5 +37,12 @@ cover:
 	@echo "+ $@"
 	@go list -f '{{if len .TestGoFiles}}"go test -coverprofile={{.Dir}}/.coverprofile {{.ImportPath}}"{{end}}' $(shell go list ${PKG}/... | grep -v vendor) | xargs -L 1 sh -c
 
+coveralls:
+	@echo "+ $@"
+# Make sure goveralls is installed.
```
Comment (nikhiljindal) [inline Makefile]: Format the comments to align with the commands.

Comment (G-Harmon) [reply Makefile]: When I indented the comment, then it was "executed" during 'make coveralls'. e.g. I saw "# Make sure goveralls is installed." in the terminal output...

**Comment by G-Harmon** [comment]
PTAL- now reading the token from the file again (since the secret-in-an-env isn't supported yet).

**Comment by nikhiljindal** [comment]
Thanks for the fixes @G-Harmon 
/lgtm
